solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/norm/impl/StandardNormalizer.java

```diff
@@ -51,24 +52,25 @@ public void setStd(String std) {
   }
 
   @Override
-  public void init(NamedParams params) throws NormalizerException {
-    super.init(params);
-    if (!params.containsKey("avg")) {
-      throw new NormalizerException("missing param avg");
-    }
-    if (!params.containsKey("std")) {
-      throw new NormalizerException("missing param std");
-    }
-    avg = params.getFloat("avg", 0);
-    std = params.getFloat("std", 1);
-    if (std <= 0) {
-      throw new NormalizerException("std must be > 0");
-    }
+  public float normalize(float value) {
+    return (value - avg) / std;
```

Comment:
How are we going to be handling invalid object states now with the plugin-style?  In this case if std == 0 the client will get an exception at query time, vs before they were getting an error response the moment they deployed a bad normalizer at offline/index time.